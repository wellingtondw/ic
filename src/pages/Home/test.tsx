import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { Home } from '.'
import { fireEvent } from '@testing-library/dom'

const mockGetYourSide = jest.fn()

jest.mock('../../context/side', () => {
  return {
    useSide: () => ({
      getYourSide: mockGetYourSide,
      loading: false
    })
  }
})

describe('<Home />', () => {
  it('should be able to render correctly', () => {
    const { container } = renderWithTheme(<Home />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to choose a side', () => {
    renderWithTheme(<Home />)

    const buttonEl = screen.getByRole('button', { name: /start/i })

    fireEvent.click(buttonEl)

    expect(mockGetYourSide).toHaveBeenCalledTimes(1)
  })
})
