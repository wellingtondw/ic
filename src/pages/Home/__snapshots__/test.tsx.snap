// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should be able to render correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
  font-weight: bold;
  border: 0;
  border-radius: 0.625rem;
  position: relative;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  background: #2C97D1;
  color: #FFFFFF;
}

.c3:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c3:hover:not(:disabled) {
  background: #298ec4;
}

.c5 {
  visibility: visible;
}

.c0 {
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  padding: 0 3rem;
}

.c1 {
  font-size: 2.25rem;
  color: #2C97D1;
  font-weight: 400;
  margin-bottom: 0.5rem;
}

.c2 {
  text-transform: uppercase;
  font-size: 0.75rem;
  color: #2C97D1;
  margin-bottom: 162px;
  -webkit-letter-spacing: 0.35rem;
  -moz-letter-spacing: 0.35rem;
  -ms-letter-spacing: 0.35rem;
  letter-spacing: 0.35rem;
  line-height: 18px;
}

.c4 {
  text-transform: uppercase;
  -webkit-letter-spacing: 0.35rem;
  -moz-letter-spacing: 0.35rem;
  -ms-letter-spacing: 0.35rem;
  letter-spacing: 0.35rem;
}

@media (min-width:480px) {
  .c3 {
    padding: 1rem 2rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding-top: 210px;
    min-height: auto;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 4.5rem;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Welcome to 
    <strong>
      iClinic
    </strong>
  </h1>
  <p
    class="c2"
  >
    Frontend Challenge
  </p>
  <button
    class="c3 c4"
  >
    <div
      class="c5"
    >
      Start
    </div>
  </button>
</div>
`;
